packages/order-experience-api/src/api/createController.ts

```diff
@@ -76,12 +82,17 @@ export class CreateController extends AbstractController<typeof requestSchema> {
       customer,
       ...calculateTotalsFromItems({ items }),
     })
-    const paymentFilterData = await savePaymentFiltersAdmin(paymentFilters)
 
-    const dto = new Data({
-      ...orderData,
-      paymentFilters: paymentFilterData,
-    })
-    return this.created<any>(res, dto.serialize())
+    if (!!paymentFilters && paymentFilters.length > 0) {
```

Comment:
Could this be like this?

`if (!!paymentFilters && paymentFilters.length > 0) {
  const paymentFiltersData = await savePaymentFiltersAdmin(paymentFilters);
}

const dto = new Data({
  ...orderData,
  paymentFilters: paymentFiltersData ? paymentFiltersData : [],
});

return this.created < any > (res, dto.serialize());`

Reply:
No need to make logic nested, lets try to keep it simple as possible,